File: app/src/main/java/com/coppermobile/mysampleusingmvvmlivedata/data/source/local/CommentsDAO.java
package com.coppermobile.mysampleusingmvvmlivedata.data.source.local;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.coppermobile.mysampleusingmvvmlivedata.data.Comments;

import java.util.List;

@Dao
public interface CommentsDAO {

    @Insert
    void insertComments(Comments Comment);

    @Query("SELECT * FROM comments_table WHERE dish_id=:dishId ORDER BY comment_time DESC;")
    LiveData<List<Comments>> getComments(int dishId);

    @Delete
    void deleteComments(Comments comments);
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/data/source/local/CommentsDao.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.data.source.local

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Comments

@Dao
interface CommentsDao {

    @Insert
    fun insertComment(comments: Comments)

    @Delete
    fun deleteComment(comments: Comments)

    @Query("select * from comments_table where dish_id=:dishId")
    fun getComments(dishId: Int): LiveData<List<Comments>>
}